### PR TITLE
Test job template that defines a test job

### PR DESCRIPTION
**Pull request by llxia**
- initial template is for test job auto generation

Close: #890

Signed-off-by: lanxia <lan_xia@ca.ibm.com>

### CODE REVIEW

**Comment by llxia** [comment]
This is the initial version of the template. The job name is created based on openj9 test job naming convention. 

As mentioned in #890, we are going to refine this. For example, use folder vs view, create the same naming convention, etc.

**Comment by llxia** [comment]
[Test_Job_Auto_Gen](https://ci.eclipse.org/openj9/view/Test/job/Test_Job_Auto_Gen/) is created at OpenJ9 Jenkins. All test jobs are auto-generated in OpenJ9 Jenkins https://ci.eclipse.org/openj9/view/Test/

File: buildenv/jenkins/testJobTemplate [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+println "SPECS: ${SPECS}"
+
+def SPEC_JENKINS_FILE_MAP = [ 
+	'aix_ppc-64_cmprssptrs' : 'openjdk_ppc64_aix',
+	'linux_ppc-64_cmprssptrs_le' : 'openjdk_ppc64le_linux',
+	'linux_390-64_cmprssptrs' : 'openjdk_s390x_linux',
+	'zos_390-64_cmprssptrs' : 'openjdk_s390x_zos',
+	'linux_x86-64_cmprssptrs' : 'openjdk_x86-64_linux',
+	'linux_x86-64' : 'openjdk_x86-64_linux_largeHeap',
+	'osx_x86-64_cmprssptrs' : 'openjdk_x86-64_osx',
```
Comment (smlambert) [inline buildenv/jenkins/testJobTemplate]: There is also an osx_x86-64_cmprssptrs_largeHeap build I believe

Comment (smlambert) [reply buildenv/jenkins/testJobTemplate]: Also a sparc solaris build coming very shortly (openjdk_sparc_solaris), but I guess we can update this as part of that PR.

File: buildenv/jenkins/testJobTemplate [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+if (binding.hasVariable('JDK_VERSIONS')) {
+	JDK_VERSIONS = JDK_VERSIONS.split(',')
+} else {
+	JDK_VERSIONS = [8,11,12]
+}
+
+if (binding.hasVariable('GROUPS')) {
+	GROUPS = GROUPS.split(',')
+} else {
+	GROUPS = ['functional','system']
```
Comment (smlambert) [inline buildenv/jenkins/testJobTemplate]: Is purpose of GROUPS used for job naming and BUILD_LIST hint?
Fallback to just these 2 groups, functional, system, is for ci.eclipse.org/openj9 builds?

File: buildenv/jenkins/testJobTemplate (hunk 1 above)
Comment (llxia) [reply buildenv/jenkins/testJobTemplate]: > There is also an osx_x86-64_cmprssptrs_largeHeap build I believe

Yes, you are right. This map was created and tested in openj9 and openj9 does not build osx_x86-64_largeHeap. I will add it.

File: buildenv/jenkins/testJobTemplate (hunk 2 above)
Comment (llxia) [reply buildenv/jenkins/testJobTemplate]: I will add the rest as the default.
yes, ci.eclipse.org/openj9 builds only run functional and system atm.

**Comment by llxia** [comment]
Changeset is updated.

**Pull request merged**